plugins/eddn.py

```diff
@@ -35,121 +46,147 @@
 this.marketId = None
 this.commodities = this.outfitting = this.shipyard = None
 
+HORIZ_SKU = 'ELITE_HORIZONS_V_PLANETARY_LANDINGS'
+
 
-class EDDN(object):
+# TODO:
+# 1, this does a bunch of magic checks for whether or not the given commander has horizons. We can get that directly
+# from  CAPI via commander.capabilities.Horizons -- Should this be changed to use that? did such a thing exist when
+# this was written?
+#
+# 2, a good few of these methods are static or could be classmethods. they should be created as such.
 
-    ### SERVER = 'http://localhost:8081'	# testing
+class EDDN:
+    # SERVER = 'http://localhost:8081'	# testing
     SERVER = 'https://eddn.edcd.io:4430'
-    UPLOAD = '%s/upload/' % SERVER
-    REPLAYPERIOD = 400	# Roughly two messages per second, accounting for send delays [ms]
-    REPLAYFLUSH = 20	# Update log on disk roughly every 10 seconds
-    TIMEOUT= 10	# requests timeout
-    MODULE_RE = re.compile('^Hpt_|^Int_|Armour_', re.IGNORECASE)
+    UPLOAD = f'{SERVER}/upload/'
+    REPLAYPERIOD = 400  # Roughly two messages per second, accounting for send delays [ms]
+    REPLAYFLUSH = 20  # Update log on disk roughly every 10 seconds
+    TIMEOUT = 10  # requests timeout
+    MODULE_RE = re.compile(r'^Hpt_|^Int_|Armour_', re.IGNORECASE)
     CANONICALISE_RE = re.compile(r'\$(.+)_name;')
 
-    def __init__(self, parent):
-        self.parent = parent
+    def __init__(self, parent: tk.Tk):
+        self.parent: tk.Tk = parent
         self.session = requests.Session()
-        self.replayfile = None	# For delayed messages
-        self.replaylog = []
+        self.replayfile: Optional[TextIO] = None  # For delayed messages
+        self.replaylog: List[str] = []
 
-    def load(self):
+    def load(self) -> bool:
         # Try to obtain exclusive access to the journal cache
```

Comment:
Is "journal cache" correct?  I'd assume this will cache *all* EDDN messages, and some of that data is from CAPI.

Reply:
It appears to be. The code path the `cmdr_data` callback takes does not touch any of the replaylog stuff